proto/wg/cosmo/graphqlmetrics/v1/graphqlmetrics.proto

```diff
@@ -0,0 +1,48 @@
+// https://protobuf.dev/programming-guides/style/
+
+syntax = "proto3";
+
+package wg.cosmo.graphqlmetrics.v1;
+
+import "wg/cosmo/common/common.proto";
+
+message SchemaUsageInfo {
+  string OperationDocument = 1;
+  repeated TypeFieldUsageInfo TypeFieldMetrics = 2;
+  OperationInfo OperationInfo = 3;
+  RequestInfo RequestInfo = 4;
+  map<string,string> Attributes = 6;
+}
+
+message OperationInfo {
+  string OperationHash = 1;
+  string OperationName = 2;
+  string OperationType = 3;
+}
+
+message RequestInfo {
+  // FederatedGraphID and OrganizationID are transport over JWT
+  string RouterConfigVersion = 3;
+}
+
+message TypeFieldUsageInfo {
+  repeated string Path = 1;
+  repeated string TypeNames = 2;
+  repeated string SourceIDs = 3;
+  uint64 Count = 4;
+}
+
+message TypeFieldSource {
+  string SubgraphID = 1;
+}
+
+message PublishGraphQLRequestMetricsRequest {
+  repeated SchemaUsageInfo SchemaUsage = 1;
+}
+
+message PublishOperationCoverageReportResponse {
+}
+
+service GraphQLMetricsService {
+  rpc PublishGraphQLMetrics(PublishGraphQLRequestMetricsRequest) returns (PublishOperationCoverageReportResponse) {}
```

Comment:
I think we should distinguish between "registering" a Document with TypeFieldUsageInfo and publishing the actual usage with attributes, etc.
Sending the Document as a string for each request will quickly become a problem, even with compression.
You've added a "Count" value, but this will only count up if the attributes are the same, right?
In addition, do you also want to report the request IDs? This could make batching worse because of cardinality.

Reply:
One way you could solve this is by adding one more rpc endpoint that only accepts the Operation Hash and the Document. When publishing the Schema Usage, you simply omit the Document and only publish the Hash. WDYT?

Reply:
All good points. Need to think about it a bit longer.

Reply:
I refactored the aggregation logic. Now, only exact same schema usage information is aggregated. Please check.
For the second comment please see https://github.com/wundergraph/cosmo/pull/174#discussion_r1359881346